src/Framework/Process.cxx

```diff
@@ -411,18 +443,55 @@ bool Process::process(int n, Event& event) const {
                    << t.AsString("lc") << ")";
   }
 
+  if (performance_) performance_->start(performance::Callback::process, 0);
   try {
+    std::size_t i_proc{0};
     for (auto module : sequence_) {
+      i_proc++;
+      if (performance_) performance_->start(performance::Callback::process, i_proc);
       if (dynamic_cast<Producer *>(module)) {
         (dynamic_cast<Producer *>(module))->produce(event);
       } else if (dynamic_cast<Analyzer *>(module)) {
         (dynamic_cast<Analyzer *>(module))->analyze(event);
       }
+      if (performance_) performance_->end(performance::Callback::process, i_proc);
     }
   } catch (AbortEventException &) {
+    if (performance_) {
+      performance_->end(performance::Callback::process, 0);
+      performance_->end_event(false);
+    }
```

Comment:
Do we want to end the process callback for the processor that threw `AbortEventException`? Right now it doesn't (which I think makes sense), but I could forsee someone wanting to know how quick aborted events take.

Reply:
Yes, we do get an `AbortEventException` from our Simulator when we abort a G4 event.
(They are thrown with the `abortEvent` protected member function of processors.)

I will update the code to `end` the timer for the processor that threw `AbortEventException`.